src/poliastro/czml/extract_czml.py

```diff
@@ -100,6 +100,52 @@ def _init_orbit_packet_cords_(self, i, rtol):
 
         return cart_cords
 
+    def _init_groundtrack_packet_cords_(self, i, rtol):
+        """
+
+        Parameters
+        ----------
+        i: int
+            Index of referenced orbit
+        rtol: float
+            Maximum relative error permitted
+
+        Returns
+        -------
+        coordinate list
+        """
+        cart_cords = []  # type: List[float]
+
+        h = (self.end_epoch - self.orbits[i][2]).to(u.second) / self.orbits[i][1]
+
+        # Get rounding factor given the relative tolerance
+        rf = 0
+        while rtol < 1:
+            rtol *= 10
+            rf += 1
+
+        ellipsoid = self.cust_prop[0]
+
+        for k in range(self.orbits[i][1] + 2):
+            position = propagate(self.orbits[i][0], TimeDelta(k * h), rtol=rtol)
```

Comment:
If I understand correctly, perhaps we could replace this `for` loop by `orbit.sample`, or perhaps propagating along the whole time interval?

Reply:
We could use ``orbit.sample`` but there is not way moving around the ``for`` loop unless we use some kind of mapping (which would make the code more hard to understand). Is there any specific advantage to using ``orbit.sample``?

Reply:
If it's going to make the code harder to understand, then no :)